gnuradio-runtime/lib/basic_block.cc

```diff
@@ -188,7 +188,7 @@ void basic_block::insert_tail(pmt::pmt_t which_port, pmt::pmt_t msg)
     msg_queue_ready[which_port]->notify_one();
 
     // wake up thread if BLKD_IN or BLKD_OUT
-    global_block_registry.notify_blk(alias());
+    global_block_registry.notify_blk(d_symbol_name);
```

Comment:
I like this change. I hope it'll improve the situation.
For some reason the block interaction with `global_block_registry` is based on `alias()`. see:
https://github.com/gnuradio/gnuradio/blob/5a2983534fa1faac49528aa41c6cacad806a70d8/gnuradio-runtime/lib/block.cc#L601
https://github.com/gnuradio/gnuradio/blob/5a2983534fa1faac49528aa41c6cacad806a70d8/gnuradio-runtime/lib/block_registry.cc#L50
I went through `gnuradio-runtime`. Though, `alias()` is used in many places. Mostly with logging and RPC. As fas as I can tell, these are the only occurences that might interact with your PR.

I assume @bastibl  and @marcusmueller are more familiar with the details.

Reply:
OK, i agree these need to change. If blocks are registered according to their `symbol_name` instead of `alias`, the call you mention in block_registry.cc becomes unnecessary entirely as it is already removed from the map here:

https://github.com/gnuradio/gnuradio/blob/5a2983534fa1faac49528aa41c6cacad806a70d8/gnuradio-runtime/lib/block_registry.cc#L48

Though this does give me some hesitation that I am missing something...

Reply:
On second thought, i believe the block_registry alias() call is still required and this area is unrelated to the modification of how blocks are notified. the ref_map gets additions here

https://github.com/gnuradio/gnuradio/blob/master/gnuradio-runtime/lib/block_registry.cc#L70

and here

https://github.com/gnuradio/gnuradio/blob/master/gnuradio-runtime/lib/block_registry.cc#L89

both of which should be cleaned up in `block_registry::block_unregister()` I believe.

PR has been updated accordingly.